src/gui/layout/qgslayoutviewtooladdnodeitem.cpp

```diff
@@ -190,16 +191,61 @@ void QgsLayoutViewToolAddNodeItem::moveTemporaryNode( QPointF scenePoint, Qt::Ke
 
 void QgsLayoutViewToolAddNodeItem::setRubberBandNodes()
 {
-  if ( QGraphicsPolygonItem *polygonItem = dynamic_cast<QGraphicsPolygonItem *>( mRubberBand.get() ) )
+  QList<QGraphicsItem *> items = mRubberBand->childItems();
+  if ( items.isEmpty() )
+    return;
```

Comment:
This is potentially an api break if a custom python based item is returning a QGraphicsPolygonItem / QGraphicsPathItem directly, instead of a group item (since in that case there won't be any children).

I think you could avoid this by eg putting the `if ( QGraphicsPolygonItem *polygonItem = dynamic_cast<QGraphicsPolygonItem *>( item) ) .... etc ` logic inside a lambda, and then if there's no child items directly call it on mRubberBand